### ISSUE TITLE
[bug] exception when using AutotoolsToolchain for Windows

### ISSUE THREAD

**Issue by dvirtz** [conan-io/conan#10139]
I'm trying to build a recipe using `AutotoolsToolchain` for Windows using mingw but get the error 
```
  File "/root/.conan/data/libpciaccess/0.16/uplynk/testing/export/conanfile.py", line 44, in generate
    tc = AutotoolsToolchain(self)
  File "/opt/pyenv/versions/3.9.7/lib/python3.9/site-packages/conan/tools/gnu/autotoolstoolchain.py", line 52, in __init__
    self._host = _get_gnu_triplet(os_host, arch_host)
  File "/opt/pyenv/versions/3.9.7/lib/python3.9/site-packages/conan/tools/gnu/get_gnu_triplet.py", line 14, in _get_gnu_triplet
    raise ConanException("'compiler' parameter for 'get_gnu_triplet()' is not specified and "
conans.errors.ConanException: 'compiler' parameter for 'get_gnu_triplet()' is not specified and needed for os=Windows
```

Looks like `AutotoolsToolchain` should pass the compiler to `get_gnu_triplet`

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04
  * Compiler+version: w64-mingw
  * Conan version: 1.42.0
  * Python version: 3.9.7

**Comment by franramirez688** [comment]
Hi @dvirtz,

I'm not sure if I'm getting the problem.

As far as I can see in the code, yes, there is a constraint:

```python
    if os_ == "Windows" and compiler is None:
        raise ConanException("'compiler' parameter for 'get_gnu_triplet()' is not specified and "
                             "needed for os=Windows")
```
It's also true that there is another check depending on the compiler below that code:

```python
    # Calculate the OS
    if compiler == "gcc":
        windows_op = "w64-mingw32"
    elif compiler == "Visual Studio":
        windows_op = "windows-msvc"
    else:
        windows_op = "windows"
```

You can have a look deeper at the code [here](https://github.com/conan-io/conan/blob/develop/conan/tools/gnu/get_gnu_triplet.py).

What I mean is that this is expected behaviour in case you're not defining a compiler inside of your settings so, do you mean the bug is that you already have the compiler defined or Conan should not request any compiler there?

**Comment by franramirez688** [comment]
@dvirtz 

You're right, sorry, `AutotoolsToolchain` is not passing the compiler, I just realized about it. So, yes, definitely, it's a bug. Good catch!

**Comment by dvirtz** [comment]
thanks @franramirez688. you beat me to it

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10141 merged, to be released in 1.44
